src/test/java/ru/training/at/hw3/tests/ExerciseFirst.java

```diff
@@ -0,0 +1,128 @@
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        Properties prop = new Properties();
+        try {
+            prop.load(fileInput);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        HomePage homePage = PageFactory.initElements(webDriver, HomePage.class);
+        WaitActions waitActions = new WaitActions(webDriver);
+
+        //1.Open site
+        homePage.openSite();
+
+        //2.Assert Browser title
+        softAssert.assertEquals(homePage.getBrowserTitle(),
+                prop.getProperty("homePageTitle"));
+
+        //3.Perform login
+        waitActions.waitUntilCondition(webDriver ->
+                homePage.getLoginButton().isDisplayed());
+        homePage.login();
+
+        //4.Assert Username is loggined
+        waitActions.waitUntilCondition(webDriver ->
+                homePage.getUserLoginName().isDisplayed());
+        softAssert.assertEquals(homePage.getUserLoginName().getText(),
+                prop.getProperty("userLoginName"));
+
+        //5.Assert that there are 4 items on the header
+        //section are displayed and they have proper texts
+        Header header = PageFactory.initElements(webDriver, Header.class);
+        waitActions.waitUntilCondition(webDriver -> header.getHomeHeader().isDisplayed());
+        softAssert.assertEquals(header.getHomeHeader().getText(),
+                prop.getProperty("homeHeader"));
+        waitActions.waitUntilCondition(webDriver -> header.getContactFormHeader().isDisplayed());
+        softAssert.assertEquals(header.getContactFormHeader().getText(),
+                prop.getProperty("contactFormHeader"));
+        waitActions.waitUntilCondition(webDriver -> header.getServiceHeader().isDisplayed());
+        softAssert.assertEquals(header.getServiceHeader().getText(),
+                prop.getProperty("serviceHeader"));
+        waitActions.waitUntilCondition(webDriver -> header.getMetalsAndColorsHeader().isDisplayed());
+        softAssert.assertEquals(header.getMetalsAndColorsHeader().getText(),
+                prop.getProperty("metalsAndColorsHeader"));
+
+        //6.Assert that there are 4 images on the Index Page and they are displayed
+        waitActions.waitUntilCondition(webDriver -> homePage.getImageFirstHomePage().isDisplayed());
+        softAssert.assertTrue(homePage.getImageFirstHomePage().isDisplayed());
+        waitActions.waitUntilCondition(webDriver -> homePage.getImageSecondHomePage().isDisplayed());
+        softAssert.assertTrue(homePage.getImageSecondHomePage().isDisplayed());
+        waitActions.waitUntilCondition(webDriver -> homePage.getImageThirdHomePage().isDisplayed());
+        softAssert.assertTrue(homePage.getImageThirdHomePage().isDisplayed());
+        waitActions.waitUntilCondition(webDriver -> homePage.getImageFourthHomePage().isDisplayed());
+        softAssert.assertTrue(homePage.getImageFourthHomePage().isDisplayed());
```

Comment:
Why do you do not combined images to list of web elements and do not use loop here for the assertion?